api/lib/oauth/oauth.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+module.exports = function(configFilePath) {
+  var request = require("request");
+  var logger = require('../log/logger');
+  var path = require('path');
+  var fs = require('fs');
+  var settings = require(path.join(__dirname, './../config/setting.js'))((JSON.parse(
+    fs.readFileSync(configFilePath, 'utf8'))));
+  var HttpStatus = require('http-status-codes');
+  var obj = {};
+  var prefix = "/v1/apps/";
+  obj.checkUserAuthorization = function(req, callback) {
+    var reqPath = req.path;
+    var startIndex = reqPath.indexOf(prefix) + prefix.length;
+    var endIndex = reqPath.indexOf("/", startIndex);
+    var appId = reqPath.substring(startIndex, endIndex);
+    if (!appId) {
+      logger.error("Failed to get appId");
+      callback({ "message": "Failed to get appId" }, null);
+      return;
+    }
+    var ccEndpoint = getCloudControllerEndpoint(req);
+    var userToken = req.header("Authorization");
+    if (!userToken) {
+      logger.info("User token is not provided");
+      callback(null, false);
+      return;
+    }
+    getUserInfo(req, function(error, userId) {
+      if (error) {
+        callback(error, null);
+      } else {
+        var options = {
+          url: ccEndpoint + "/v2/users/" + userId + "/spaces?q=app_guid:" + appId + "&q=developer_guid:" + userId,
+          method: "GET",
+          json: true,
+          timeout: 10000,
+          headers: {
+            "Authorization": userToken,
+            "Content-Type": "application/json"
+          }
+        };
+        request(options, function(error1, response, body) {
+          if (error1) {
+            logger.error("Failed to get space with userId and AppId during permission check", { "appId": appId, "userId": userId, "error": error1 });
+            callback(error1, null);
+          } else {
+            if (response.statusCode == HttpStatus.OK) {
+              var totalResults = body.total_results;
+              if (totalResults > 0) {
+                logger.info("Permission check passed", { "appId": appId, "userId": userId })
+                callback(null, true);
+              } else {
+                logger.info("Permission check not passed", { "appId": appId, "userId": userId });
+                callback(null, false);
+              }
+
+            } else {
+              var errorObj = {
+                "statusCode": response.statusCode
+              };
+              logger.error("Failed to get space with userId and AppId during permission check", { "appId": appId, "userId": userId, "error": errorObj });
+              callback(errorObj, null);
+            }
+          }
+        });
+      }
+
+
+    });
+
+  }
+
+  function getCloudControllerEndpoint(req) {
+    return settings.cfApi;;
+  };
+
+  function getCloudFoundryInfo(req, callback) {
```

Comment:
Is that unnecessary to pass-through "req" here?  the cc endpoint is set through settings as defined in func getCloudControllerEndpoint, not generated from "req"